entrypoint.sh

```diff
@@ -190,21 +190,70 @@ function start_postfix {
 
     postfix start
 
+    # migrating older single-domain DKIM (/var/db/dkim/default.*) to /var/db/dkim/$HOSTNAME/default.*
+    if [ -f /var/db/dkim/default.private ]; then
+        echo "Migrating ${HOSTNAME} keys to /var/db/dkim/$HOSTNAME/"
+        mkdir -p /var/db/dkim/$HOSTNAME
+        mv /var/db/dkim/default.* /var/db/dkim/$HOSTNAME
+        chmod 400 /var/db/dkim/$HOSTNAME/default.private
+        chown opendkim:opendkim /var/db/dkim/$HOSTNAME/default.private
+    fi
 
-    # DKIM
-    if [ ! -f /var/db/dkim/default.private ]; then
-        mkdir -p /var/db/dkim
-        echo "OpenDKIM: Keys not found, generating..."
-        opendkim-genkey -b 2048 -d $HOSTNAME -D /var/db/dkim/ -s default -v
+    echo "Inserting ${HOSTNAME} data to /etc/opendkim/{KeyTable, SigningTable, TrustedHosts}"
+    echo "default._domainkey.${HOSTNAME} ${HOSTNAME}:default:/var/db/dkim/${HOSTNAME}/default.private" >> /etc/opendkim/KeyTable
+    echo "${HOSTNAME} default._domainkey.${HOSTNAME}" >> /etc/opendkim/SigningTable
+    echo "${HOSTNAME}" >> /etc/opendkim/TrustedHosts
 
-        chmod 400 /var/db/dkim/default.private
-        chown opendkim:opendkim /var/db/dkim/default.private
+    for virtualDomain in $virtualDomains; do
+        # skip generating keys for $HOSTNAME twice in case it is also used as forwarded domain.
+        if [ "$virtualDomain" = "$HOSTNAME" ]; then
+            continue
+        fi
```

Comment:
```suggestion
    allDomains="$virtualDomains"'
    [[ $virtualDomains =~ $HOSTNAME ]] || {
        virtualDomains="$virtualDomains $HOSTNAME"
    }
    for virtualDomain in $allDomains; do
```

Reply:
Very nice! And it does not manipulate `virtualDomains`, solving cases like mine.

Reply:
@petslane just to confirm, there is a typo on line 202 (extra `'`). I'm asking because, you never know with bash exoteric stuff.

Reply:
Yes, this is typo. `'` key too close to `enter` key.